### PR TITLE
Updated Bootstrap Datepicker Definitely Typed Definitions

### PR DESCRIPTION
**Pull request by rposener**
Please fill in this template.

- [ X] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

If changing an existing definition:
- [X ] Provide a URL to documentation or source code which provides context for the suggested changes: https://bootstrap-datepicker.readthedocs.io/en/latest/options.html
- [ X] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by sheetalkamat** [comment]
Thanks for the contribution. This PR contains merge conflicts and cannot be merged at the moment. Please refresh this PR and address the merge conflicts.

**Comment by rposener** [comment]
Fixed up - should merge now.

**Comment by sheetalkamat** [comment]
@borisyankov can you please review. Thanks

**Comment by mhegazy** [comment]
@brobo

File: types/bootstrap-datepicker/index.d.ts [hunk 1]
```diff
@@ -17,32 +17,57 @@
 interface DatepickerOptions {
     format?: string | DatepickerCustomFormatOptions;
     weekStart?: number;
-    startDate?: any;
-    endDate?: any;
+    startDate?: Date|string;
```
Comment (aozgaa) [inline types/bootstrap-datepicker/index.d.ts]: Please add spaces between union type constituents.

File: types/bootstrap-datepicker/index.d.ts [hunk 2]
```diff
@@ -17,32 +17,57 @@
...
+    endDate?: Date|string;
     autoclose?: boolean;
     startView?: number;
-    todayBtn?: any;
+    todayBtn?: boolean|"linked";
     todayHighlight?: boolean;
     keyboardNavigation?: boolean;
     language?: string;
-    beforeShowDay?: (date: any) => any;
+    beforeShowDay?: (date: Date) => undefined|string|boolean|DatepickerBeforeShowDayResponse;
```
Comment (aozgaa) [inline types/bootstrap-datepicker/index.d.ts]: rename `DatepickerBeforeShowDayResponse` to `DatepickerBeforeShowResponse`.

File: types/bootstrap-datepicker/index.d.ts [hunk 3]
```diff
@@ -17,32 +17,57 @@
...
+
+interface DatepickerViewDate {
+    year:number;
+    /** Month starting with 0 */
+    month:number;
+    /** Day of the month starting with 1 */
+    day:number;
+}
+
+interface DatepickerBeforeShowResponse {
```
Comment (aozgaa) [inline types/bootstrap-datepicker/index.d.ts]: make this
``` ts
type DatepickerBeforeShowResponse = undefined | string | boolean | { enabled?: boolean; classes?: string; tooltip?: string; }

**Pull request merged**